packages/block/block.ts

```diff
@@ -0,0 +1,248 @@
+/* eslint-disable @typescript-eslint/unbound-method */
+import {
+  childNodes$,
+  cloneNode$,
+  createEventListener,
+  insertBefore$,
+  insertText,
+  innerHTML$,
+  remove$ as removeElement$,
+  setAttribute,
+  setText,
+  mapGet$,
+  mapHas$,
+  mapSet$,
+  setStyleAttribute,
+  setSvgAttribute,
+  firstChild$,
+  nextSibling$,
+} from './dom';
+import { renderToTemplate } from './template';
+import { AbstractBlock } from './types';
+import type { Edit, EditChild, Props, VElement, Hole } from './types';
+
+export const createBlock = (fn: (props?: Props) => VElement) => {
+  const cache = new Map<string, Hole>();
+  const vnode = fn(
+    new Proxy(
+      {},
+      {
+        // A universal getter will return a Hole instance if props[any] is accessed
+        // Allows code to identify holes in virtual nodes ("digs" them out)
+        get(_, key: string): Hole {
+          if (mapHas$.call(cache, key)) return mapGet$.call(cache, key);
+          const hole = { __key: key };
+          mapSet$.call(cache, key, hole);
+          return hole;
```

Comment:
Also, if `Hole` is not going to be complicated, do we really need a cache here?

Reply:
probably not. good point here